pandas/tests/frame/indexing/test_column_vs_index_tz.py

```diff
@@ -0,0 +1,57 @@
+import pandas as pd
+import pandas._testing as tm
+
+
+class TestColumnvsIndexTZEquality:
+    # https://github.com/pandas-dev/pandas/issues/29463
+    def check_for_various_tz(self, tz):
+        df = pd.DataFrame(
+            {
+                "val": range(10),
+                "time": pd.date_range(start="2019-01-01", freq="1d", periods=10, tz=tz),
+            }
+        )
+        df_query = df.query('"2019-01-03 00:00:00+00" < time')
+        l1 = pd.DataFrame(list(df_query["time"]))
```

Comment:
Is this is the "expected" output, we'll want to construct this with a different method without query

```
expected = pd.DataFrame(...)
```

Reply:
@mroeschke  ,    
I modified everything as per suggestion and the tests pass locally.
But here instead of constructing new "expected" DataFrame we would want to use the ones already generating by using query since "We are testing for result to be same at index and column level"
This is what was expected as per issue ? Right ?

Reply:
We want to make sure the "expected" DataFrame takes an independent path than query so if `query` breaks both "result" and "expected" don't change.

So want to make sure the result of:
`result = df.set_index('time').query('"2019-01-03 00:00:00+00" < time')`

gives a result of:
`expected = pd.DataFrame(...)`

Reply:
When we create an expected dataframe , it passes for few tz and fails for others , bcoz the query string always expects the time zone to be "UTC" , query('"2019-01-03 00:00:**00+00" < time')**, and therefore there is shape mismatch  for example for "Asia/Kolkata" it is failing whereas it passes for "US/Eastern".

I assume we cannot change the query string because that is what we are testing for , in that case knowing the expected o/p when comparing to different tz will result in shape mismatch which is exactly happening now.
Any thoughts ?

```
def test_check_column_vs_index_tz_query(self, tz_aware_fixture):
    # https://github.com/pandas-dev/pandas/issues/29463

    tz = tz_aware_fixture
    expected = pd.DataFrame(
        {"time": pd.date_range(start="2019-01-04", freq="1d", periods=7, tz=tz)}
    ).set_index("time")

    df = pd.DataFrame(
        {"time": pd.date_range(start="2019-01-01", freq="1d", periods=10, tz=tz)}
    )
    result = df.set_index("time").query('"2019-01-03 00:00:00+00" < time')
    tm.assert_frame_equal(result, expected)
```

```
AssertionError: DataFrame are different

DataFrame shape mismatch
[left]:  (7, 0)
[right]: (8, 0)
```

Reply:
Since we're just testing that we can query with a tz offset in the query string when the index is tz-aware, we can be with what the query string is.

You can just change the test to be then

```
    tz = tz_aware_fixture
    expected = pd.DataFrame(index=pd.date_range(start="2019-01-04", freq="1d", periods=10, tz=tz, name='time'))

    df = pd.DataFrame(index=pd.date_range(start="2019-01-01", freq="1d", periods=10, tz=tz, name=time))
    result = df.query('"2018-01-03 00:00:00+00" < time')
    tm.assert_frame_equal(result, expected)
```

Reply:
@mroeschke , Yes this works , I am now banging my head for not thinking this earlier.
Thank you very much , now all the tests pass.

As suggested , I have added the test in test_query_eval.py file.